mlflow/server/js/src/components/tables/EditableFormTable.js

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import { Table, Input, Form, Icon } from 'antd';
+import PropTypes from 'prop-types';
+
+import './EditableFormTable.css';
+
+const EditableContext = React.createContext();
+
+class EditableCell extends React.Component {
+  static propTypes = {
+    editing: PropTypes.bool,
+    dataIndex: PropTypes.string,
+    title: PropTypes.string,
+    record: PropTypes.object,
+    index: PropTypes.number,
+    children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
+    save: PropTypes.func,
+    cancel: PropTypes.func,
+    recordKey: PropTypes.string,
+  };
+
+  handleKeyPress = (event) => {
+    const { save, recordKey, cancel } = this.props;
+    if (event.key === 'Enter') {
+      save(recordKey);
+    } else if (event.key === 'Escape') {
+      cancel();
+    }
+  };
+
+  render() {
+    const { editing, dataIndex, title, record, children, ...restProps } = this.props;
+    return (
+      <EditableContext.Consumer>
+        {({ getFieldDecorator }) => (
+          <td {...restProps} className={editing ? 'editing-cell' : ''}>
```

Comment:
This line results in a React warning:

```
bundle.js:420976 Warning: Invalid values for props `save`, `cancel` on <td> tag. Either remove them from the element, or pass a string or number value to keep them in the DOM. For details, see https://fb.me/react-attribute-behavior
    in td (at EditableFormTable.js:36)
    in EditableCell (created by TableCell)
    in TableCell (created by TableRow)
    in tr (created by BodyRow)
    in BodyRow (created by TableRow)
    in TableRow (created by Connect(TableRow))
    in Connect(TableRow) (created by ExpandableRow)
    in ExpandableRow (created by Connect(ExpandableRow))
    in Connect(ExpandableRow) (created by BaseTable)
    in tbody (created by BaseTable)
    in table (created by BaseTable)
    in BaseTable (created by Connect(BaseTable))
    in Connect(BaseTable) (created by BodyTable)
    in div (created by BodyTable)
    in BodyTable (created by ExpandableTable)
```

Reply:
Filtering out `save` and `cancel` from `restProps` seems to fix it (since the `save` & `cancel` props don't actually need to be set on this td)

Reply:
Actually here I have already explicitly set className so no need to dump restProps.